apps/src/templates/teacherDashboard/LoginTypePicker.jsx

```diff
@@ -40,33 +71,22 @@ class LoginTypePicker extends Component {
   };
 
   render() {
-    const {title, providers, setLoginType, handleCancel, disabled} = this.props;
+    const {providers, setLoginType, handleCancel, disabled} = this.props;
     const withGoogle =
       providers && providers.includes(OAuthSectionTypes.google_classroom);
     const withMicrosoft =
       providers && providers.includes(OAuthSectionTypes.microsoft_classroom);
     const withClever =
       providers && providers.includes(OAuthSectionTypes.clever);
     const hasThirdParty = withGoogle | withMicrosoft | withClever;
-
-    // explicitly constrain the container as a whole to the width of the
-    // content. We expect that differing length of translations versus english
-    // source text can cause unexpected layout changes, and this constraint
-    // should help mitigate some of them.
-    const containerStyle = {maxWidth: styleConstants['content-width']};
-
-    // anchor email address policy note to footer just above 'Cancel' button
-    const emailPolicyNoteStyle = {
-      position: 'absolute',
-      top: '0px',
-      zIndex: '600'
-    };
+    // Adjust max height of the dialog based on number of LoginCard types available to the user
+    style.root.maxHeight = hasThirdParty ? '500px' : '360px';
```

Comment:
This is so the dialog isn't too big right? Maybe we shouldn't be setting a `height` at all then and let the content take up as much space at it needs. We should probably have a `minHeight` for screens which are too small though. Then we wouldn't need this ternary operator; we would just need a single min-height.
Doing this kind of stuff is fragile because someone might change the card sizes in the future, but then forget to update this `maxHeight`.

Maybe I'm misunderstanding this though.

Reply:
I added it as more of an aesthetic thing so that the dialog doesn't have too large of a blank area when there's only one row of login types, the scrolling still works regardless of what the height is set at since that's just the maxHeight of the full dialog not for the height of the scrolling portion. I'm happy to remove the maxHeight and make it consistent regardless of the number of login types. What do you think would work best?

Reply:
I agree that we should avoid a big white gap should be avoided. Looking at the CSS, I think the reason there is a gap is because `height` is being set to `80vh`. I'm wondering if we can remove that `height` and `maxHeight`, then let the cards dynamically adjust the height as they are added. Consider what should happen if in 2 years we have 7 logins rather than 5. Then there would be a 3rd row. With the current solution I think that would result in a scroll bar appearing even if there is enough screen real estate.

Reply:
I'll need to keep the heights to a degree because otherwise it doesn't scroll properly due to the CardContainer using a FlexBox. I'll look into if there's fewer of them that I can use to have slightly more dynamically-determined dimensions so the height works better and I'll let you know.